app/controllers/students_controller.rb

```diff
@@ -0,0 +1,56 @@
+class StudentsController < ApplicationController
+  before_action :find_student, only: %i(show edit update destroy)
+
+  def index
+    @students = Student.all
+  end
+
+  def show; end
+
+  def new
+    @student = Student.new
+  end
+
+  def create
+    @student = Student.new student_params
+
+    if @student.save
+      flash[:info] = t(".success")
+      redirect_to root_url
+    else
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    if @student.update student_params
+      flash[:success] = t ".update"
+      redirect_to @student
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @student.destroy
```

Comment:
destroy failed thì sao

Reply:
- thông thường thì ko failed, nếu failed thì thêm flash thông báo, redirect to students thôi
- cái này pull sau (form delete) mới xem tiếp, pull này chỉ "add student".